notes/zrq/20210219-01-live-deployment.txt

```diff
@@ -0,0 +1,198 @@
+            ./bin/zeppelin-daemon.sh restart
+
+        popd
+
+
+# -----------------------------------------------------
+# Use the timing logger to check the performance.
+#[user@desktop]
+
+    gaiauser=$(secret aglais.zeppelin.gaiauser)
+    gaiapass=$(secret aglais.zeppelin.gaiapass)
+
+    cookiejar=/tmp/cookies
+    zeppelinurl=http://zeppelin.aglais.uk:8080
+
+    curl \
+        --silent \
+        --request 'POST' \
+        --cookie-jar "${cookiejar:?}" \
+        --data "userName=${gaiauser:?}" \
+        --data "password=${gaiapass:?}" \
+        "${zeppelinurl:?}/api/login" \
+    | jq '.'
+
+    >   {
+    >     "status": "OK",
+    >     "message": "",
+    >     "body": {
+    >       "principal": "gaiauser",
+    >       "ticket": "4342bdd9-0a7b-4f27-a216-bae695a69b22",
+    >       "roles": "[\"role1\"]"
+    >     }
+    >   }
+
+
+
+    noteident=2FYSBR6FT
+
+    curl \
+        --silent \
+        --cookie "${cookiejar:?}" \
+        "${zeppelinurl:?}/api/notebook/${noteident:?}" \
+    | jq '.' \
+    | sed '
+        /"dateStarted": null,/d
+        /"dateStarted":/ {
+            h
+            s/\([[:space:]]*\)"dateStarted":[[:space:]]*\("[^"]*"\).*$/\1\2/
+            x
+            }
+        /"dateFinished": null,/ d
+        /"dateFinished":/ {
+            H
+            x
+            s/[[:space:]]*"dateFinished":[[:space:]]*\("[^"]*"\).*$/ \1/
+            s/\([[:space:]]*\)\(.*\)/\1echo "\1\\"elapsedTime\\": \\"$(datediff --format "%H:%M:%S" --input-format "%b %d, %Y %H:%M:%S %p" \2)\\","/e
+            x
+            G
+            }
+        ' \
+    | jq '
+        .body.paragraphs[] | {
+            title:  .title,
+            status: .status,
+            result: .results.code,
+            time:   .elapsedTime,
+            output: ((.results | select(.msg | length > 0) | .msg[] | select(.type == "TEXT") | .data | split("\n") | map(select(startswith("-")))) // "-")
+            }
+        '
+
+    >   {
+    >     "title": null,
+    >     "status": "FINISHED",
+    >     "result": "SUCCESS",
+    >     "time": "0:0:2",
+    >     "output": "-"
+    >   }
+    >   ....
+    >   ....
+    >   {
+    >     "title": null,
+    >     "status": "READY",
+    >     "result": null,
+    >     "time": null,
+    >     "output": "-"
+    >   }
+
+    #
+    # Notebooks are specific to each user.
+    # gaiauser created 2FYZ4R8FE
+    # zrq created 2FYSBR6FT
+    #
+
+    #
+    # If zrq tries to access a note created by gaiauser
+
+        Insufficient privileges to read note.
+        Allowed users or roles: [gaiauser]
+        But the user zrq belongs to: [zrq, user]
+
+    #
+    # Much more to learn and document about how Zeppelin manages multiple-users.
+    # Learn about what Livy offers
+
+    # Suggest we change /user/zrq .. to /home/zrq ...
+    # Closer to what end users would expect ?
+
+    # Can we hide the fedora user ?
+    # Set home to somewhere else in /etc/password
+
+
+
+
```

Comment:
Does this live deploy not include the integration with Github?

Reply:
Nope, the current live deploy does not have the GitHub integration enabled.